Made strobe width of HWPE Stream Fifo a parameter

Lynx005F opened the pull request:
Simply moved all the strobe with calculation into a parameter so that it could be externally defined if needed.

This change is needed for redundant RedMulE as data size will be increased with parity bits while Strobe stays the same.

FrancescoConti commented:
Seems ok. But it would be better to align to the same approach everywhere, not only in the FIFO - can you take care of that?

Lynx005F commented:
Sure! Check the newest commit. I also added pass-trough to lower level modules.
Key point to check are:
- The merge and split modules - IMO should be good like this.
- The modules that have a hardcoded width of 32 internally e.g. stream source,
I am unsure  if there is a usecase for having it there. For example for my usecase it wouldn't make any sense:
   - I increase the data width by 1 to accomodate a parity bit while keeping the strobe width the same since now 33 bits are effectively a unit.
